src/pmdas/bpf/modules/exitsnoop.c

```diff
@@ -0,0 +1,468 @@
+        if (offset == iter) {
+            *val = i;
+            return true;
+        }
+        iter++;
+    }
+    return false;
+}
+
+#define INDOM_COUNT 2
+static unsigned int indom_id_mapping[INDOM_COUNT];
+
+#define METRIC_COUNT 9
+enum metric_name { AGE, PID, TID, PPID, SIG, EXIT_CODE, COMM, COREDUMP, LOST };
+enum metric_indom { EXITSNOOP_INDOM, LOST_EVENTS };
+
+char* metric_names[METRIC_COUNT] = {
+    [AGE]        =  "exitsnoop.age",
+    [PID]        =  "exitsnoop.pid",
+    [TID]        =  "exitsnoop.tid",
+    [PPID]       =  "exitsnoop.ppid",
+    [SIG]        =  "exitsnoop.sig",
+    [EXIT_CODE]  =  "exitsnoop.exit_code",
+    [COMM]       =  "exitsnoop.comm",
+    [COREDUMP]   =  "exitsnoop.coredump",
+    [LOST]       =  "exitsnoop.lost",
+};
+
+char* metric_text_oneline[METRIC_COUNT] = {
+    [AGE]        =  "the AGE of the process with hundredth of a second resolution",
+    [PID]        =  "Process identifier",
+    [TID]        =  "Thread ID",
+    [PPID]       =  "The process that will be notified",
+    [SIG]        =  "The signal",
+    [EXIT_CODE]  =  "The exit code for exit() or the signal number for a fatal signal.",
+    [COMM]       =  "The process command name",
+    [COREDUMP]   =  "bool value repressed if it's coredump or not",
+    [LOST]       =  "The Number of the lost events",
+};
+
+char* metric_text_long[METRIC_COUNT] = {
+    [AGE]        =  "the AGE of the process with hundredth of a second resolution",
+    [PID]        =  "Process identifier",
+    [TID]        =  "Thread ID",
+    [PPID]       =  "The process that will be notified",
+    [SIG]        =  "The signal",
+    [EXIT_CODE]  =  "The exit code for exit() or the signal number for a fatal signal.",
+    [COMM]       =  "The process command name",
+    [COREDUMP]   =  "bool value repressed if it's coredump or not",
+    [LOST]       =  "The Number of the lost events",
+};
+
+static unsigned int exitsnoop_metric_count()
+{
+    return METRIC_COUNT;
+}
+
+static char* exitsnoop_metric_name(unsigned int metric)
+{
+    return metric_names[metric];
+}
+
+static unsigned int exitsnoop_indom_count(void)
+{
+    return INDOM_COUNT;
+}
+
+static void exitsnoop_set_indom_serial(unsigned int local_indom_id, unsigned int global_id)
+{
+    indom_id_mapping[local_indom_id] = global_id;
+}
+
+static int exitsnoop_metric_text(int item, int type, char **buffer)
+{
+    if (type & PM_TEXT_ONELINE) {
+        *buffer = metric_text_oneline[item];
+    } else {
+        *buffer = metric_text_long[item];
+    }
+
+    return 0;
+}
+
+static void exitsnoop_register(unsigned int cluster_id, pmdaMetric *metrics, pmdaIndom *indoms)
+{
+    /* bpf.exitsnoop.age */
+    metrics[AGE] = (struct pmdaMetric)
+    {
+        .m_desc = {
+            .pmid  = PMDA_PMID(cluster_id, 0),
+            .type  = PM_TYPE_DOUBLE,
+            .indom = indom_id_mapping[EXITSNOOP_INDOM],
+            .sem   = PM_SEM_INSTANT,
+            .units = PMDA_PMUNITS(0, 0, 0, 0, 0, 0),
```

Comment:
PCP metric metadata includes the metric value units. In this case we can tell the PCP client tools the metric value is in nanoseconds (`PM_TIME_NSEC`).
Example in the runqlat module: https://github.com/performancecopilot/pcp/blob/7adaaedd665d947d94b5e7acd045f7fbb53d127c/src/pmdas/bpf/modules/runqlat.c#L71
Detailed docs: https://man7.org/linux/man-pages/man3/pmLookupDesc.3.html

Reply:
The second parameter to `PMDA_PMUNITS` needs to be `1` to specify that it's a time unit.